ESP8266Interface.cpp

```diff
@@ -64,20 +67,31 @@ int ESP8266Interface::connect(const char *ssid, const char *pass, nsapi_security
 
 int ESP8266Interface::connect()
 {
-    _esp.setTimeout(ESP8266_CONNECT_TIMEOUT);
-    
-    if (!_esp.reset()) {
```

Comment:
Hang on, so there's now no reset in a connect, only on initialisation.

Is that wise? For general stability, it seems like we should keep resetting on each connect. (But obviously not before we check we're not already connected)

Reply:
[About resetting UBLOX](https://github.com/ARMmbed/mbed-os/issues/5582#issuecomment-347542454). So you want that reset is run unless connection has already been established. I'll fix this.

Reply:
Well, it should always reset when it actually connects, if that's what's worked until now.

An attempt to connect when already connected should be an error that does nothing.